streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -714,6 +728,39 @@ public void close(boolean clean,
         taskClosed = true;
     }
 
+    private boolean retrieveCommittedTimestamp(final TopicPartition partition) {
```

Comment:
Why is the return type not `void` (similar for `setAssignmentToStoredTimestamps` below)? Seem you added it for testing? I would prefer to keep it `void` and change the tests if possible.